drivers/platform/x86/asus-wmi.c

```diff
@@ -1619,6 +1625,25 @@ static int is_display_toggle(int code)
 	return 0;
 }
 
+static bool asus_wmi_is_fnlock_enabled(struct asus_wmi *asus)
```

Comment:
When I first saw this I interpreted `is_fnlock_enabled` to mean "Fn-Lock is currently active".
But actually I see that the hardware doesn't let you query this, and this function is instead checking if the Fn-Lock device and functionality exists on the platform.
I suggest renaming this to `asus_wmi_has_fnlock_key` or similar for clarity.